openstack/container/experimental/capsules/results.go

```diff
@@ -0,0 +1,106 @@
+package capsules
+
+import (
+	"encoding/json"
+	"time"
+
+	"github.com/gophercloud/gophercloud"
+)
+
+type commonResult struct {
+	gophercloud.Result
+}
+
+// Extract is a function that accepts a result and extracts a capsule resource.
+func (r commonResult) Extract() (*Capsule, error) {
+	var s *Capsule
+	err := r.ExtractInto(&s)
+	return s, err
+}
+
+// GetResult represents the result of a get operation.
+type GetResult struct {
+	commonResult
+}
+
+// Represents a Container Orchestration Engine Bay, i.e. a cluster
+type Capsule struct {
+	// UUID for the capsule
+	UUID string `json:"uuid"`
+
+	// ID for the capsule
+	ID int `json:"id"`
+
+	// User ID for the capsule
+	UserID string `json:"user_id"`
+
+	// Project ID for the capsule
+	ProjectID string `json:"project_id"`
+
+	// cpu for the capsule
+	CPU float64 `json:"cpu"`
+
+	// Memory for the capsule
+	Memory string `json:"memory"`
+
+	// The name of the capsule
+	MetaName string `json:"meta_name"`
+
+	// Indicates whether capsule is currently operational. Possible values include:
+	// Running,
+	Status string `json:"status"`
+
+	// The created time of the capsule.
+	CreatedAt time.Time `json:"-"`
+
+	// The updated time of the capsule.
+	UpdatedAt time.Time `json:"-"`
+
+	// Links includes HTTP references to the itself, useful for passing along to
+	// other APIs that might want a server reference.
+	Links []interface{} `json:"links"`
+
+	// The capsule version
+	CapsuleVersion string `json:"capsule_version"`
+
+	// The capsule restart policy
+	RestartPolicy string `json:"restart_policy"`
+
+	// The capsule metadata labels
+	MetaLabels map[string]string `json:"meta_labels"`
+
+	// The list of containers uuids inside capsule.
+	ContainersUUIDs []string `json:"containers_uuids"`
+
+	// The capsule IP addresses
+	Addresses map[string][]Address `json:"addresses"`
+
+	// The capsule volume attached information
+	VolumesInfo map[string][]string `json:"volumes_info"`
```

Comment:
Similar to `Addresses`, this should probably be `map[string][]Volume`. However, the Zun documentation says that volume support is incomplete. If this is true, then `[]string` is fine for now.

Reply:
Actually, I apologize. The above comment is assuming that `VolumesInfo` is returning an object/hash/struct about Volumes (such as name and ID). However, if it is only returning a list of UUIDs (similar to `ContainerUUIDs`, then ignore the above comment altogether.

Reply:
Volume support for Zun doesn't support Cinder multiple Attach. For one-one attach, this works fine. Besides, Zun will add volume multiple attach recently. So the volumes_info data structure will be compatible with volume attach.
The structure of VolumesInfo is:
"volumes_info": {
              volume1_uuid: [container1_uuid, container2_uuid],
              volume2_uuid: [container1_uuid, container3_uuid],
              volume3_uuid: [container3_uuid, container4_uuid]
}
So for the key volume uuid, it will record all the containers that attached to them.

For addresses, there is a new [patch](https://review.openstack.org/542604) added subnet yesterday. I will add corresponding support.